### PR TITLE
Binomial ideals

### PR DESCRIPTION
**Pull request by CarloSircana**

### CODE REVIEW

File: src/Rings/mpoly-ideals.jl [hunk 1]
```diff
@@ -10,9 +10,8 @@ export ngens, gens
 @doc Markdown.doc"""
     ideal(g::Array{T, 1}) where {T <: MPolyElem}
 
-Create the ideal generated by the polynomials in `g`. 
-
-# Examples
+Creates the ideal generated by the polynomials in `g`. 
+###### Example
```
Comment (fingolfin) [inline src/Rings/mpoly-ideals.jl]: The Julia documentation suggest that docstrings should uniformly use `# Examples` to section off examples (even if it just a single example, the plural is kept). I suggest we keep following that guidelines (it has the advantage that one does not have to remember to add/remove an "s" in a section when adding/removing examples)

Comment (CarloSircana) [reply src/Rings/mpoly-ideals.jl]: Sorry, I must have done something wrong while merging the code. I will fix it.

File: experimental/Rings/QabAndPChars.jl [hunk 2]
```diff
@@ -518,197 +517,157 @@ function quadratic_irrationality_info(a::QabModule.QabElem)
...
 
+function isroot_of_unity(a::QabElem)
+  return istorsion_unit(a.data)
+  #=
+  b = a^a.c
+  return b.data == 1 || b.data == -1
+  =#
+end
+
+function Oscar.order(a::QabElem)
```
Comment (thofma) [inline experimental/Rings/QabAndPChars.jl:528]: Could you add a comment on what this is doing? What is the order of an element?

Comment (CarloSircana) [reply experimental/Rings/QabAndPChars.jl:528]: I will add a comment, but it was already there. This is a first implementation of the abelian closure of Q that Claus did and it was just done to provide what the algorithms for binomial ideals needed.

File: src/Rings/mpoly-ideals.jl [hunk 3]
```diff
@@ -237,17 +233,15 @@ function saturation_with_index(I::MPolyIdeal, J::MPolyIdeal)
...
 # elimination #######################################################
 @doc Markdown.doc"""
     eliminate(I::MPolyIdeal, l::Array{MPolyElem, 1})
 
-Given a list of polynomials which are variables, these variables are eliminated from `I`. 
-
-That is, return the ideal of all polynomials in `I` which only depend on the remaining variables.
-
+Given a list of polynomials which are variables, these variables are eliminated from `I`. That is,
+the function returns the ideal of all polynomials in `I` which only depend on the remaining variables.
```
Comment (fingolfin) [inline src/Rings/mpoly-ideals.jl]: I think @wdecker arranged these doc strings in this way (with two paragraphs) on purpose. Is there a reason to change that, or is it more fallout from the bad merge? Same elsewhere in this PR

Comment (fingolfin) [reply src/Rings/mpoly-ideals.jl]: @CarloSircana if you tell me these docstring changes are not intended, I can update your PR to remove them with <3 minutes of work, using some `git` kungfu. Just saying this as I know cleaning this up manually can be arduous, and I don't know how proficient you are with git. If you think you can do it yourself with just a few minutes effort, of course just do it; but I don't want you to waste half an hour on my nitpicks :-)

Comment (CarloSircana) [reply src/Rings/mpoly-ideals.jl]: Yes, I didn't intend to change them, so if you could do it it would help a lot :)

Comment (fingolfin) [reply src/Rings/mpoly-ideals.jl]: Done now. Please check the diff (I ended up squashing everything into a single commit, so you may need to discard your local version of the branch and fetch it fresh from the server)

File: experimental/Rings/QabAndPChars.jl (hunk 2 above)
Comment (thofma) [reply experimental/Rings/QabAndPChars.jl:528]: Did you add the comment?

**Comment by thofma** [comment]
There is something messed up with the documentation build, but it is unrelated to these changes here.

**Pull request merged**